go/pkg/tool/tool.go

```diff
@@ -326,18 +326,45 @@ func (c *Client) DownloadDirectory(ctx context.Context, rootDigest, path string)
 	return err
 }
 
+// UploadStats contains various metadata of a directory upload.
+type UploadStats struct {
```

Comment:
optional nit: technically since it's a main package, it doesn't need to be exported.

Reply:
It's not, though. This is the `tool` package.

Reply:
I think it'd still work like a factory pattern, allowing the main package to access the struct when returned from here, but disallowing it from creating new instances of this struct.

Reply:
Why? It's dead data. What are you trying to achieve?

Reply:
I guess my point was that: not exporting this struct would still work, even if it's used in an external package like `main`.

Reply:
I misread the PR, and I'd recommend to not change anything here. A simple exported struct was the right choice IMO.